pkg/client/userd/trafficmgr/traffic_manager.go

```diff
@@ -544,24 +544,37 @@ func (tm *TrafficManager) WorkloadInfoSnapshot(
 	ctx context.Context,
 	namespaces []string,
 	filter rpc.ListRequest_Filter,
+	allNamespaces bool,
 	includeLocalIntercepts bool,
 ) (*rpc.WorkloadInfoSnapshot, error) {
 	tm.WaitForNSSync(ctx)
 	tm.wlWatcher.waitForSync(ctx)
-
-	nss := make([]string, 0, len(namespaces))
-	for _, ns := range namespaces {
-		ns = tm.ActualNamespace(ns)
-		if ns != "" {
-			nss = append(nss, ns)
+	nss := make([]string, 0)
+	nsmap := make(map[string]bool)
+	is := tm.getCurrentIntercepts()
+	if allNamespaces {
+		// get unique set of namespaces from active intercepts
+		for _, intercept := range is {
+			if _, value := nsmap[intercept.Spec.Namespace]; !value {
+				nsmap[intercept.Spec.Namespace] = true
+				nss = append(nss, intercept.Spec.Namespace)
+			}
+		}
+	} else {
+		// validate user-provided namespaces
+		for _, ns := range namespaces {
+			ns = tm.ActualNamespace(ns)
+			if ns != "" {
+				nss = append(nss, ns)
+			}
+		}
+		if len(nss) == 0 {
```

Comment:
It strikes me as odd that this check was moved into the else-clause. Why isn't it valid in the `allNamspaces` case? Surely there will be cases when there are no intercepts? If the objective is to list all current intercepts regardless of namespace, then why continue here?